File: src/main/java/org/jdc/template/ui/IndividualEditActivity.java
package org.jdc.template.ui;

import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;

import org.jdc.template.R;
import org.jdc.template.dagger.Injector;
import org.jdc.template.event.IndividualSavedEvent;
import org.jdc.template.event.RxBus;

import javax.inject.Inject;

import butterknife.Bind;
import butterknife.ButterKnife;
import pocketknife.BindExtra;
import pocketknife.PocketKnife;

public class IndividualEditActivity extends BaseActivity {

    public static final String EXTRA_ID = "INDIVIDUAL_ID";

    @Bind(R.id.ab_toolbar)
    Toolbar toolbar;
    @BindExtra(EXTRA_ID)
    long individualId;

    @Inject
    RxBus bus;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.fragment_single);
        Injector.get().inject(this);
        ButterKnife.bind(this);
        PocketKnife.bindExtras(this);

        setSupportActionBar(toolbar);
        enableActionBarBackArrow(true);

        setDefaultKeyMode(DEFAULT_KEYS_SEARCH_LOCAL);

        setupActionBar();

        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.fragment_pos1, IndividualEditFragment.newInstance(individualId))
                    .commit();
        }
    }

    @Override
    protected void onStart() {
        super.onStart();
        addSubscription(bus.subscribeMainThread(event -> handleSubscribeMainThread(event)));
    }

    private void setupActionBar() {
        setSupportActionBar(toolbar);
        ActionBar actionBar = getSupportActionBar();

        if (actionBar != null) {
            actionBar.setTitle(R.string.individual);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item == null) {
            return false;
        }

        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    private void handleSubscribeMainThread(Object event) {
        if (event instanceof IndividualSavedEvent) {
            finish();
        }
    }
}

File: src/main/kotlin/org/jdc/template/ui/IndividualEditActivity.kt
package org.jdc.template.ui

import android.app.Activity
import android.os.Bundle
import android.view.MenuItem
import kotlinx.android.synthetic.main.toolbar_actionbar.*
import org.jdc.template.R
import org.jdc.template.dagger.Injector
import org.jdc.template.event.IndividualSavedEvent
import pocketbus.Bus
import pocketbus.Subscribe
import pocketbus.SubscriptionRegistration
import pocketbus.ThreadMode
import pocketknife.BindExtra
import javax.inject.Inject

class IndividualEditActivity : BaseActivity() {

    @BindExtra(EXTRA_ID)
    var individualEditId: Long = 0

    @Inject
    lateinit var bus: Bus

    lateinit var subscriptionRegistration: SubscriptionRegistration

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.fragment_single)
        Injector.get().inject(this)

        // temp fix for kapt issue (https://youtrack.jetbrains.com/issue/KT-9183)
        individualEditId = getIntent().getLongExtra(EXTRA_ID, 0);
//        PocketKnife.bindExtras(this)

        setSupportActionBar(ab_toolbar)
        enableActionBarBackArrow(true)

        setDefaultKeyMode(Activity.DEFAULT_KEYS_SEARCH_LOCAL)

        setupActionBar()

        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction().add(R.id.fragment_pos1, IndividualEditFragment.newInstance(individualEditId)).commit()
        }
    }

    override fun onStart() {
        super.onStart()
        subscriptionRegistration = bus.register(this)
    }

    override fun onStop() {
        bus.unregister(subscriptionRegistration)
        super.onStop()
    }

    private fun setupActionBar() {
        setSupportActionBar(ab_toolbar)
        val actionBar = supportActionBar

        actionBar?.setTitle(R.string.individual)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item == null) {
            return false
        }

        when (item.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
            else -> return super.onOptionsItemSelected(item)
        }
    }

    @Subscribe(ThreadMode.MAIN)
    fun handleIndividualSavedEvent1(event: IndividualSavedEvent) {
        finish()
    }

    companion object {

        const val EXTRA_ID = "INDIVIDUAL_ID"
    }
}